src/indieweb_utils/indieauth/profile.py

```diff
@@ -0,0 +1,80 @@
+from dataclasses import dataclass
+
+import requests
+
+from bs4 import BeautifulSoup
+
+
+class ProfileError(Exception):
+    pass
+
+
+@dataclass
+class Profile:
+    name: str
+    photo: str
+    url: str
+    email: str
+
+
+def get_profile(
+        me: str
+    ) -> Profile:
+
+    try:
+        me_profile = requests.get(me)
+    except:
+        raise ProfileError("Request to retrieve profile URL did not return a valid response.")
+
+    if not me_profile:
+        return Profile(
+            name="",
+            photo="",
+            url="",
+            email=""
+        )
```

Comment:
In what case would me_profile be `None`? I'd expect it would always exist because errors are raising a ProfileError above. Can probably delete this?